packages/crusher-electron-app/src/extension/v2/ui/components/Sidebar/Steps.tsx

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import { Text } from "@dyson/components/atoms/text/Text";
+import { Checkbox, CheckboxProps } from "@dyson/components/atoms/checkbox/checkbox";
+import { css } from "@emotion/react";
+import { MoreIcon } from "crusher-electron-app/src/extension/assets/icons";
+import { TextBlock } from "@dyson/components/atoms/textBlock/TextBlock";
+import { Conditional } from "@dyson/components/layouts";
+import { Dropdown } from "@dyson/components/molecules/Dropdown";
+
+export const containerStyle = css`
+	padding: 26rem;
+`;
+
+export function Steps(): JSX.Element {
+	const [s, setS] = React.useState([]);
+	const [selected, setSelected] = React.useState(new Set());
+
+	const toggleStep = React.useCallback(
+		(index) => {
+			const selectedSteps = new Set(selected);
+			console.log(selected);
+			if (selected.has(index)) {
+				selectedSteps.delete(index);
+			} else {
+				selectedSteps.add(index);
+			}
+
+			setSelected(selectedSteps);
+		},
+		[selected],
+	);
+
+	return (
+		<div
+			css={css`
+				border-top: 1rem solid #303235;
+			`}
+		>
+			<div
+				css={css`
+					display: flex;
+					align-items: center;
+					padding: 14rem 26rem;
+				`}
+			>
+				<Checkbox isSelected={!!s.length} callback={(a) => setS([a])} />
```

Comment:
Selecting this should select all the steps, and deselecting it should deselect all the steps. If the user selects all the steps manually it should automatically be selected